src/stages/stage5/spells/atmospheric_discharge.c

```diff
@@ -10,61 +10,61 @@
 
 #include "spells.h"
 
-MODERNIZE_THIS_FILE_AND_REMOVE_ME
-
-void iku_atmospheric(Boss *b, int time) {
-	if(time < 0) {
-		GO_TO(b, VIEWPORT_W/2+200.0*I, 0.06);
-		return;
+TASK(cloud_shoot, NO_ARGS) {
+	for(int x = 0;; x += WAIT(difficulty_value(6, 5, 4, 3))) {
+		if(global.diff >= D_Hard) {
+			PROJECTILE(
+				.proto = pp_thickrice,
+				.pos = VIEWPORT_W * rng_real(),
+				.color = RGB(0, 0.3, 0.7),
+				.move = move_accelerated(0, 0.01 * I),
+			);
+		} else {
+			PROJECTILE(
+				.proto = pp_rice,
+				.pos = VIEWPORT_W * rng_real(),
+				.color = RGB(0, 0.3, 0.7),
+				.move = move_linear(2 * I),
+			);
+		}
 	}
+}
 
-	int t = time % 500;
-	TIMER(&t);
-
-	GO_TO(b,VIEWPORT_W/2+tanh(sin(time/100))*(200-100*(global.diff==D_Easy))+I*VIEWPORT_H/3+I*(cos(t/200)-1)*50,0.03);
-
-	FROM_TO(0, 500, 23-2*global.diff) {
-		tsrand_fill(4);
-		cmplx p1 = VIEWPORT_W*afrand(0) + VIEWPORT_H/2*I*afrand(1);
-		cmplx p2 = p1 + (120+20*global.diff)*cexp(0.5*I-afrand(2)*I)*(1-2*(afrand(3) > 0.5));
-
-		int i;
-		int c = 6+global.diff;
+TASK(ball_shoot, NO_ARGS) {
+	int difficulty = difficulty_value(1, 2, 3, 4);
+	for(int x = 0;; x += WAIT(difficulty_value(19, 17, 15, 13))) {
+		RNG_ARRAY(rand, 4);
+		cmplx p1 = VIEWPORT_W * vrng_real(rand[0]) + VIEWPORT_H/2 * I * vrng_real(rand[1]);
+		cmplx p2 = p1 + (120 + 20 * difficulty) * cdir(0.5 - vrng_real(rand[2])) * (1 - 2 * (vrng_real(rand[3]) > 0.5));
```

Comment:
`+ difficulty_value(140, 160, 180, 200)` or `+20*difficulty_value(7,8,9,10)`

`cdir(0.5 - vrng_real(rand[2]))` -> `cdir(0.5*vrng_sreal(rand[2]))`

for the last one maybe `vrng_chance(rand[3], 0.5)` is better?